examples/llama2_driver.py

```diff
@@ -25,66 +20,83 @@ def _get_world_size() -> int:
     return int(os.environ.get("LOCAL_WORLD_SIZE", "1"))
 
 def _load_model(args, precision=torch.bfloat16):
+    rank = _get_rank()
+    world_size = _get_world_size()
 
-    if args.pipeline_parallel:
-        rank = _get_rank()
-        world_size = _get_world_size()
+    config = ModelArgs.from_name(Path(args.checkpoint_path).parent.name)
+    config.n_local_layers = config.n_local_layers // world_size
 
-        torch.cuda.set_device(rank)
-        dist.init_process_group(backend="nccl", rank=rank, world_size=world_size)
-        model_path = f'models/model_{rank:02d}.pt'
-    else:
-        torch.cuda.set_device(args.device)
-        model_path = f'models/model.pt'
+    with torch.device('meta'):
```

Comment:
What is 'meta' device?

Reply:
It is something to do with being able to create arbitrarily large tensors. May not be required for this specific example though.

Reply:
then let's remove it

Reply:
Done!